typescript/sdk/src/router/types.ts

```diff
@@ -97,12 +98,32 @@ export const ForeignDeploymentConfigSchema = z.object({
   foreignDeployment: z.string().optional(),
 });
 
-export const RemoteRouterDomain = z.string();
+export const RemoteRouterDomainOrChainNameSchema = z.string().or(z.number());
+export type RemoteRouterDomainOrChainName = z.infer<
+  typeof RemoteRouterDomainOrChainNameSchema
+>;
+
+export function resolveRouterMapConfig<T>(
+  multiProvider: MultiProvider,
+  routerMap: Record<RemoteRouterDomainOrChainName, T>,
+): Record<number, T> {
+  return Object.fromEntries(
+    Object.entries(routerMap).map(([domainIdOrChainName, value]) => {
+      if (isNumeric(domainIdOrChainName)) {
+        return [parseInt(domainIdOrChainName), value];
+      }
+
+      const meta = multiProvider.getChainMetadata(domainIdOrChainName);
+      return [meta.domainId, value];
```

Comment:
```suggestion
    Object.entries(routerMap).map(([chain, value]) => 
      [multiProvider.getChainMetadata(chain).domainId, value]
```
I think this is equivalent